pkg/cmd/stack/deploy.go

```diff
@@ -634,15 +631,19 @@ func validateDefinedServices(s *model.Stack, servicesToDeploy []string) error {
 	return nil
 }
 
-func validateDependentServiceDeployed(s *model.Stack, servicesToDeploy []string) error {
-	for _, svcToDeploy := range servicesToDeploy {
-		for name := range s.Services[svcToDeploy].DependsOn {
-			if !isSvcToBeDeployed(servicesToDeploy, name) {
-				return fmt.Errorf("service '%s' depends on '%s' to be deployed but its not deployed", svcToDeploy, name)
+func addDependentServicesIfNotPresent(ctx context.Context, s *model.Stack, options *StackDeployOptions, c kubernetes.Interface) {
+	added := make([]string, 0)
+	for _, svcToDeploy := range options.ServicesToDeploy {
+		for dependentSvc := range s.Services[svcToDeploy].DependsOn {
+			if !isSvcToBeDeployed(options.ServicesToDeploy, dependentSvc) && !isSvcRunning(ctx, s.Services[dependentSvc], s.Namespace, dependentSvc, c) {
```

Comment:
if it is running, does the redeploy logic work?
we could add a test when A depends on B, and we just redeploy A

Reply:
No, only services that are in the `ServicesToDeploy` list are deployed/redeployed. 
I have covered the following scenarios on tests:
- Job not running (deploy), failed(redeploy) and completed or active (do nothing)
- Deployment not running (deploy) and running (do nothing)
- Statefulset not running (deploy) and running (do nothing)